cmd/jaeger/internal/integration/e2e_integration.go

```diff
@@ -109,7 +110,24 @@ func (s *E2EStorageIntegration) e2eInitialize(t *testing.T, storage string) {
 			return false
 		}
 		defer resp.Body.Close()
-		return resp.StatusCode == http.StatusOK
+		if s.HealthCheckEndpoint == "" {
```

Comment:
We don't want this branching. All e2e tests now run the health extension, why do we need to special case?

Reply:
Noted, will make the change.

Reply:
I am quite not able to figure out, as to why the endpoint is saying connection refused for Kafka. It doesn't say the same for other config files?

Reply:
I don't think Kafka is starting, why would it have no logs?
<img width="882" alt="image" src="https://github.com/user-attachments/assets/0fb33c21-1199-4b6d-bde3-465a9b0b6604">

Reply:
actually this might be expected, since we do shutdown in the script
```
teardown_kafka() {
  echo "Stopping Kafka..."
  docker compose -f "${compose_file}" down
}
```
this is a bug - in other scripts we dump the logs from teardown, before we shutdown the compose env.

Reply:
another thing I do not see is the output of the logs from the two binaries you're starting, why is that?

Reply:
> actually this might be expected, since we do shutdown in the script
> 
> ```
> teardown_kafka() {
>   echo "Stopping Kafka..."
>   docker compose -f "${compose_file}" down
> }
> ```
> 
> this is a bug - in other scripts we dump the logs from teardown, before we shutdown the compose env.

@joeyyy09 PTAL at this.

Reply:
> another thing I do not see is the output of the logs from the two binaries you're starting, why is that?

I don't think I have made any changes that should affect the logs at any level. I have just changed the healthcheck URL.
That should have no effect whatsoever.

Reply:
I am not saying it's because of your changes, I am saying that we do have code in e2e struct that dumps the logs of the binaries, and I do not see this happening for Kafka tests. So this is both an operational issue and a blocker for you to troubleshoot why the health checks are failing in the test.

Reply:
Ohhh Okay.